nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -540,4 +630,72 @@ public void close() {
       api.close();
     }
   }
+
+  public void commitView(
+      ViewMetadata base,
+      ViewMetadata metadata,
+      String newMetadataLocation,
+      String contentId,
+      ContentKey key)
+      throws NessieConflictException, NessieNotFoundException {
+    UpdateableReference updateableReference = getRef();
+
+    updateableReference.checkMutable();
+
+    Branch current = (Branch) updateableReference.getReference();
+    Branch expectedHead = current;
+    if (base != null) {
+      String metadataCommitId =
+          base.properties()
+              .getOrDefault(
+                  NessieTableOperations.NESSIE_COMMIT_ID_PROPERTY, expectedHead.getHash());
+      if (metadataCommitId != null) {
+        expectedHead = Branch.of(expectedHead.getName(), metadataCommitId);
+      }
+    }
+
+    long versionId = metadata.currentVersion().versionId();
+
+    ImmutableIcebergView.Builder newViewBuilder = ImmutableIcebergView.builder();
+    // Directly casting to `SQLViewRepresentation` as only SQL type exist in
+    // `ViewRepresentation.Type`.
+    // Assuming only one engine's dialect will be used, Nessie IcebergView currently holds one
+    // representation.
+    // View loaded from catalog will have all the representation as it parses the view metadata
+    // file.
+    SQLViewRepresentation sqlViewRepresentation =
+        (SQLViewRepresentation) metadata.currentVersion().representations().get(0);
```

Comment:
Currently Nessie's `IcebergView` model stores only one dialect and representation. I didn't change Nessie side to hold multiple representation because it was needed for _global state_. Now that `global state` is removed in Nessie, only view metadata location  is enough. It will have current version id and all the representation for that version.

Reply:
it seems weird to only store one and not all of the representations

Reply:
Either Nessie can store all the representations or None I guess. But storing representations is not really required as global state is removed in Nessie, only view metadata location is enough. Lets see what others think. 

cc: @dimas-b, @snazy

Reply:
Should we even bother with storing representations in Nessie at this point? I'd say, until Nessie supports multiple representations, we can just keep metadata location and ignore all representations here. These fields are nullable in Nessie.

Reply:
True. I mentioned this above.

> Either Nessie can store all the representations or None I guess

I think we don't need to store any representation if we don't need those for metadata in Nessie project. So, I am waiting for @snazy's reply.

Reply:
I think this is the last major point that needs to be adressed before this PR can go in

Reply:
Agree.

Reply:
All metadata should really be taken directly from the view metadata - Nessie should really _not_ store the dialect nor the SQL representation. I don't understand how this related to "global state" - those things are not related to each other at all.

Also the unconditional cast is error prone IMO - similar to assuming that there is only one representation - either of different kinds or multiple SQL representation.

Reply:
Aah, `SqlText` is not nullable. So, I cannot construct without it. I will raise a PR at Nessie side to make it Nullable.

https://github.com/projectnessie/nessie/blob/fff9154da6a19cf289c09f2996b75f3cd15e0ca0/api/model/src/main/java/org/projectnessie/model/IcebergView.java#L56

Reply:
added a dummy sqlText and dummy dialect as we have concluded that we don't have to modify at Nessie side.